src/app/inventory/store/d2-item-factory.service.ts

```diff
@@ -887,6 +918,29 @@ export function D2ItemFactory(
     };
   }
 
+  function buildMasterworkInfo(
+    sockets: DimSockets,
+    statDefs: LazyDefinition<DestinyStatDefinition>
+  ): DimMasterwork | null {
+    const progress = _.find(_.pluck(sockets.sockets, 'masterworkProgress'), (mp) => mp >= 0);
+    const typeName = _.find(_.pluck(sockets.sockets, 'masterworkType'), (type) => type !== null);
+    const typeIcon = _.find(_.pluck(sockets.sockets, 'masterworkIcon'), (icon) => icon !== null);
```

Comment:
`find` is a supported ES6 method, so you can do `myList.find((icon) => icon !== null)`.

Reply:
Nice, I'll clean this up later.